comp/core/flare/helpers/send_flare.go

```diff
@@ -254,13 +260,63 @@ func SendTo(cfg pkgconfigmodel.Reader, archivePath, caseID, email, apiKey, url s
 		return "", err
 	}
 
-	r, err := readAndPostFlareFile(archivePath, caseID, email, hostname, url, source, client, apiKey)
-	if err != nil {
-		return "", err
+	// Retry logic for the actual flare file posting
+	var lastErr error
+	var defaultRetryConfig = RetryConfig{
+		MaxRetries: 3,
+		BaseDelay:  100 * time.Millisecond,
 	}
 
-	defer r.Body.Close()
-	return analyzeResponse(r, apiKey)
+	for attempt := 0; attempt <= defaultRetryConfig.MaxRetries; attempt++ {
+		if attempt > 0 {
+			// Exponential backoff
+			delay := time.Duration(attempt) * defaultRetryConfig.BaseDelay
+			time.Sleep(delay)
+		}
+
+		r, err := readAndPostFlareFile(archivePath, caseID, email, hostname, url, source, client, apiKey)
+		if err != nil {
+			if isRetryableFlareError(err) {
+				lastErr = err
+				continue // Retry
+			}
+			// Non-retryable error - return immediately
+			return "", err
+		}
+
+		if r.StatusCode >= 200 && r.StatusCode < 500 {
+			defer r.Body.Close()
+			return analyzeResponse(r, apiKey)
+		} else if r.StatusCode >= 500 && r.StatusCode < 600 {
```

Comment:
This should be in the `isRetryableFlareError` so all the retry logic is in one place.